File: src/main/java/me/lokka30/levelledmobs/rules/ModalListParsingInfo.java
package me.lokka30.levelledmobs.rules;

import me.lokka30.levelledmobs.misc.CachedModalList;
import org.jetbrains.annotations.NotNull;

import java.util.List;
import java.util.Map;

/**
 * Holds various info used for parsing rules.yml
 *
 * @author stumper66
 * @since 3.7.5
 */
public class ModalListParsingInfo {
    public ModalListParsingInfo(final @NotNull ModalListParsingTypes type){
        this.type = type;
    }

    public final ModalListParsingTypes type;
    public String configurationKey;
    public String itemName;
    public boolean supportsGroups;
    public CachedModalList<?> cachedModalList;
    public Map<String, List<String>> groupMapping;
}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/rules/ModalListParsingInfo.kt
package io.github.arcaneplugins.levelledmobs.rules

import io.github.arcaneplugins.levelledmobs.enums.ModalListParsingTypes
import io.github.arcaneplugins.levelledmobs.misc.CachedModalList

/**
 * Holds various info used for parsing rules.yml
 *
 * @author stumper66
 * @since 3.7.5
 */
class ModalListParsingInfo(
    val type: ModalListParsingTypes
) {
    var configurationKey: String? = null
    var itemName: String? = null
    var supportsGroups: Boolean = false
    var cachedModalList: CachedModalList<*>? = null
    var groupMapping: MutableMap<String, MutableSet<String>>? = null
}